### PR TITLE
[Rector] Apply Latest Rector 

### PR DESCRIPTION
**Pull request by samsonasik**
| Q             | A
| ------------- | ---
| QA      | yes

This PR apply to latest rector and update to use latest rector compatible config:

- bootstrapping files via Option::BOOTSTRAP_FILES config.
- remove deprecated cache Option:: ENABLE_CACHE config as cached by default.
- skip `RemoveUnusedPromotedPropertyRector` as phpinsights still support php 7.4.
- run the rector.

### CODE REVIEW

File: src/Application/ConfigResolver.php [hunk 1]
```diff
@@ -192,13 +192,13 @@ private static function preparePreset(array $preset, array $config): array
         $removedRulesByPreset = [];
         $addedRulesByConfig = [];
 
-        if (isset($preset['remove']) && is_array($preset['remove']) && count($preset['remove']) > 0) {
+        if (isset($preset['remove']) && is_array($preset['remove']) && $preset['remove'] !== []) {
             array_walk_recursive($preset['remove'], static function ($value) use (&$removedRulesByPreset): void {
```
Comment (50bhan) [inline src/Application/ConfigResolver.php:196]: Is there any specific reason (benefit) to change this?

**Comment by 50bhan** [comment]
@samsonasik Thank you for this update. Really appreciate you work on Rector ❤️

File: src/Application/ConfigResolver.php (hunk 1 above)
Comment (samsonasik) [reply src/Application/ConfigResolver.php:196]: It is faster

Comment (50bhan) [reply src/Application/ConfigResolver.php:196]: Ah! Saw it coming -> https://www.reddit.com/r/PHP/comments/oj1xmk/how_do_you_routinely_check_that_an_array_is_empty/

**Comment by samsonasik** [comment]
I updated to latest 0.11.48

**Review by Jibbarth — APPROVED** [review]
Thank you @samsonasik :+1:

**Pull request merged**